app/models/backend_api.rb

```diff
@@ -1,9 +1,17 @@
 # frozen_string_literal: true
 
 class BackendApi < ApplicationRecord
+  include ThreeScale::Search::Scopes
   include SystemName
   include ProxyConfigAffectingChanges::ModelExtension
 
+  self.allowed_search_scopes = %i[query]
+
+  scope :by_query, ->(query) do
+    options = {ids_only: true, per_page: 1_000_000, star: true, ignore_scopes: true, with: { }}
+    where(id: search(ThinkingSphinx::Query.escape(query), options))
+  end
+
```

Comment:
As I mentioned in the description of https://github.com/3scale/porta/pull/2204, we'll later have to polish all this and add tests 👍